src/vsr/replica.zig

```diff
@@ -5626,7 +5626,8 @@ pub fn ReplicaType(
                     .view_change => if (header.view == self.view) return,
                     else => unreachable,
                 },
-                .recovering_head => {},
+                // We need a start_view from any other replica — don't request it from ourselves.
+                .recovering_head => if (self.primary_index(header.view) == self.replica) return,
                 else => unreachable,
```

Comment:
Do we have a corresponding assert at `on_request_start_view`?

Reply:
https://github.com/tigerbeetledb/tigerbeetle/blob/0535ee33f16f2da2f46c774b3eab19bf7f797343/src/vsr/replica.zig#L1326